File: Poin B.py/PC.py
from Processor import Processor
from Disk import Disk
from RAM import RAM
class PC:
    #atribut
    
    ProcessorObj = Processor() 
    DiskObj = Disk()
    RAMObj = RAM()
    totalPrice = 0
    
    #constructor
    def __init__(self, p = Processor(), d = Disk(), r = RAM()) :
        self.ProcessorObj = p
        self.DiskObj = d
        self.RAMObj = r
        self.totalPrice = p.getPrice() + d.getPrice() + r.getPrice()
    
    
    #set get methods
    def setTotalPrice(self,  p = Processor(), d = Disk(), r = RAM()) :
        self.totalPrice =  p.getPrice() + d.getPrice() + r.getPrice()
        
    def setProcessor(self, p = Processor()) :
        self.ProcessorObj = p
        
    def setDisk(self, d = Disk()) : 
        self.DiskObj = d
        
    def setRAM(self, r = RAM()) : 
        self.RAMObj = r
        
    def getTotalPrice(self) :
        return self.totalPrice
    
    def getProcessor(self) :
        return self.ProcessorObj
    
    def getDisk(self) :
        return self.DiskObj
    
    def getRAM(self) :
        return self.RAMObj
    
    
    def output(self) :
        self.ProcessorObj.output()
        self.DiskObj.output()
        self.RAMObj.output()
        print("----------------------------------------------------+")
        print("Total Price                  : " + str(self.getTotalPrice()))
        
        
    
        
        
        
        
    
    
    

File: Poin B.php/PC.php
<?php
include "Processor.php";
include "Disk.php";
include "RAM.php";

class PC
{
    //atribute
    private $ProcessorObj;
    private $DiskObj;
    private $RAMObj;
    private $totalPrice;

    // konstruktor
    function __construct($P, $D, $R)
    {
        $this->ProcessorObj = $P;
        $this->DiskObj = $D;
        $this->RAMObj = $R;
        $this->totalPrice = $P->getPrice() + $D->getPrice() + $R->getPrice();
    }

    //get set Methods
    function  setProcessor($P)
    {
        $this->ProcessorObj = $P;
    }

    function  setDisk($D)
    {
        $this->DiskObj = $D;
    }

    function  setRam($R)
    {
        $this->RAMObj = $R;
    }
    
    function  setTotalPrice($P, $D, $R)
    {
        $this->totalPrice = $P->getPrice() + $D->getPrice() + $R->getPrice();
    }

    function getProcessor()
    {
        return $this->ProcessorObj;
    }

    function getDisk()
    {
        return $this->DiskObj;
    }

    function getRam()
    {
        return $this->RAMObj;
    }
    function getTotalPrice()
    {
        return $this->totalPrice;
    }

    // Output Method
    function  output()
    {
        $this->ProcessorObj->Output();
        $this->DiskObj->Output();
        $this->RAMObj->Output();
        echo "Total Harga               : " . $this->getTotalPrice() . "<br>";
    }
    //destructor
    function __destruct()
    {
    }
}

?>